code-challenges.test.js

```diff
@@ -10,45 +12,152 @@
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
-var nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
+// 2a) Pseudo code:
+// Copy and paste test code template into assignment (make sure nothing is inside initial template)
+// Read problem statement and understand inputs/outputs (inputs need to be directly inserted into test code)
+// Create relevant info for describe, it, expect, and toEqual methods
+// Test code output to ensure it is failing red (should get a reference error saying function is not defined)
 
+describe("minMax", () => {
+  it("A function which takes in an array of numbers and returns the min and max numbers inside a new array, in that order", () => {
+    expect(minMax([3, 56, 90, -8, 0, 23, 6])).toEqual([-8, 90]);
+    expect(minMax([109, 5, 9, 67, 8, 24])).toEqual([5, 109]);
+  });
+});
 
 // b) Create the function that makes the test pass.
 
+// 2b) Pseudo code:
+// Create a function and name it minMax (const declaration)
+// Name the parameter as array
+// Inside the function, declare a variable named minMaxArray (let declaration), and initialize it as an empty array
+// Create a line which evaluates the argument array with Math.min(), and have the result .push into minMaxArray
+// Create another line which does the same thing, but with Math.max() instead.
+// return minMaxArray
+// Run the test to verify it passes green
+
+const minMax = (array) => {
+  let minMaxArray = [Math.min(...array), Math.max(...array)];
+  //   minMaxArray.push(Math.min(...array));
+  //   minMaxArray.push(Math.max(...array));
+  return minMaxArray;
+};
 
+// Notes: I ran the test initially to ensure I got the right error message, and I did get function undefined. I then created the code, though I did initially struggle with the implementation. I didn't realize the spread operator was needed for the array argument, since Math.min/Math.max won't take in arrays as arguments. After that I got the test to pass green. I realized afterwards I could just create a one liner instead of using .push for each value. This is a nice little refactor. I don't think I'd want to refactor anymore aside from this (could maybe use sort on the argument and pick out the numbers at the smallest and largest indices). Also, I left the original two lines commented out but not deleted. Please let me know if this isn't preferred (it seems like a better way to actually show the refactoring).
```

Comment:
Great work. I like the refactor of putting the logic straight into the array. At this point you could just return the array rather than creating a variable. Make it a neat little one liner.